src/Tribe/Asset/Script_Data.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Handles adding script data to the page in cases where localizing a
+ * specific script is not suitable.
+ *
+ * Should generally be accessed via tribe( 'tribe.asset.script-data' )
+ * rather than via direct instantiation.
+ */
+class Tribe__Asset__Script_Data {
+	/**
+	 * Container for any JS data objects that should be added to the page.
+	 *
+	 * @var array
+	 */
+	protected $objects = array();
+
+	/**
+	 * Hooks up the method used to actually render the JSON data.
+	 */
+	public function hook() {
+		if ( is_admin() ) {
+			add_action( 'admin_footer', array( $this, 'render_json' ) );
+		} else {
+			add_action( 'wp_footer', array( $this, 'render_json' ) );
+		}
+	}
+
+	/**
+	 * Adds the provided data to the list of objects that should be available
+	 * to other scripts.
+	 *
+	 * @param string $object_name
+	 * @param array  $data
+	 */
+	public function add( $object_name, array $data ) {
+		$this->objects[ $object_name ] = $data;
+	}
+
+	/**
+	 * Outputs the
+	 * @internal
+	 */
+	public function render_json() {
+		if ( empty( $this->objects ) ) {
+			return;
+		}
+
+		echo '<script type=\'text/javascript\'> /* <![CDATA[ */';

```

Comment:
Doesn't seem like that's the right way to go; once this has been sent to the browser the object is in memory even if you subsequently remove the script element.

We'd need to overwrite either before it hits the browser or define a new object of the same name - separate script elements wouldn't really help.